### PR TITLE
NFDIV-1116 - Caseworker HWF Part Payment Made event

### PR DESCRIPTION
**Pull request by amoriartyMOJ**
### Change description ###

New event to allow caseworkers to record a part payment with HWF.

### JIRA link (if applicable) ###

https://tools.hmcts.net/jira/browse/NFDIV-1116

**Before merging a pull request make sure that:**

- [X] tests have been updated / new tests has been added (if needed)
- [X] README and other documentation has been updated / added (if needed)

### CODE REVIEW

File: src/main/java/uk/gov/hmcts/divorce/caseworker/event/CaseworkerHWFPartPaymentMade.java [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+    @Override
+    public void configure(final ConfigBuilder<CaseData, State, UserRole> configBuilder) {
+        new PageBuilder(configBuilder
+            .event(CASEWORKER_HWF_PART_PAYMENT_MADE)
+            .forStateTransition(EnumSet.of(AwaitingHWFPartPayment, AwaitingHWFEvidence), Submitted)
+            .name(EVENT_NAME_AND_DESCRIPTION)
+            .description(EVENT_NAME_AND_DESCRIPTION)
+            .showEventNotes()
+            .grant(CREATE_READ_UPDATE, CASE_WORKER)
```
Comment (aramchurn21) [inline src/main/java/uk/gov/hmcts/divorce/caseworker/event/CaseworkerHWFPartPaymentMade.java:39]: Hi @amoriartyMOJ I believe the super user should also have the access to trigger the event.

File: src/main/java/uk/gov/hmcts/divorce/caseworker/event/CaseworkerHWFPartPaymentMade.java [hunk 2]
```diff
@@ -0,0 +1,42 @@
...
+            .grantHistoryOnly(SOLICITOR, SUPER_USER, LEGAL_ADVISOR, CITIZEN));
+    }
```
Comment (aramchurn21) [inline src/main/java/uk/gov/hmcts/divorce/caseworker/event/CaseworkerHWFPartPaymentMade.java:41]: I think the CaseWorker should be given history access.

File: src/main/java/uk/gov/hmcts/divorce/caseworker/event/CaseworkerHWFPartPaymentMade.java (hunk 1 above)
Comment (amoriartyMOJ) [reply src/main/java/uk/gov/hmcts/divorce/caseworker/event/CaseworkerHWFPartPaymentMade.java]: Given Superuser access, thanks!

File: src/main/java/uk/gov/hmcts/divorce/caseworker/event/CaseworkerHWFPartPaymentMade.java (hunk 2 above)
Comment (amoriartyMOJ) [reply src/main/java/uk/gov/hmcts/divorce/caseworker/event/CaseworkerHWFPartPaymentMade.java:41]: caseworker has regualr access (CRU) so won't need history too

**Review by aramchurn21 — APPROVED** [review]

**Review by nitinprabhu — APPROVED** [review]

**Pull request merged**